snoop/html.py

```diff
@@ -0,0 +1,103 @@
+import re
+from bs4 import BeautifulSoup, UnicodeDammit
+import lxml.html
+import lxml.html.clean
+import lxml.etree
+
+def _extract_links(html):
+    from lxml.etree import fromstring, tostring, HTMLParser
```

Comment:
Since we're discussing pedantics, here's another minor quibble: maybe it makes more sense to `from lxml import etree` at the top of the module and then do `etree.fromstring` etc?